listeners.js

```diff
@@ -0,0 +1,166 @@
+const _ = require('lodash')
+const jsonpointer = require('jsonpointer')
+
+class Listener{
+    constructor(schema, ui_schema, formRef){
+        this.schema = Object.assign({}, schema)
+        this.ui_schema = Object.assign({}, ui_schema)
+        this.formRef = formRef
+        this.listeners = this._getListeners()
+    }
+
+    _getListeners() {
+        let acc = {}
+        this._getListenersFromUISchema(this.ui_schema, '', acc)
+        return acc
+    }
+
+    _getListenersFromUISchema(ui_schema, path, acc) {
+        const listeners = ui_schema["listener"] || ui_schema["logic:listener"]
+        if(listeners){
+            acc[path] = (acc[path] || [])
+            if (_.isArray(listeners)) {
+                acc[path].concat(listeners)
+            }
+            else{
+                acc[path].push(listeners)
+            }
+        }
+
+        if(ui_schema["fields"]) {
```

Comment:
It looks like you'd be better off extracting `ui_schema["fields"]` or `ui_schema["item"]["fields"]` to a variable and then running the `_.each` loop one time on that.